### PR TITLE
y_test, y_true args removed from roc constructor

### LINKED ISSUE #121: minor fix to ROC
I noticed that the constructor in `ROC` takes `y_true` and `y_score` as arguments, but this should not be the case. The idea is that the constructor should only take the data required for plotting. In this case, that's `fpr` and `tpr` (and they should not have default values).

The `from_raw_data` should take `y_true` and `y_score`, compute `fpr` and `tpr` and call the constructor.

The confusion probably comes from the `ConfusionMatrix` implementation. I had to do it that way since I made a release with an old API that needed change when we added the `plot_a + plot_b` feature.

### PR DESCRIPTION
**Pull request by yafimvo**
## Describe your changes
1. `y_test`, `y_true` args removed from roc constructor
2. Generates roc using `fpr` and `tpr` values
3. ROC and ROCAdd implemented with `AbstractPlot` and `AbstractComposedPlot`
4. Examples updated to `roc.from_raw_data(y_true, y_score)`
5. Old API invokes `roc.from_raw_data(y_true, y_score, ax=ax)`

## Issue ticket number and link
Closes #121 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/sklearn_evaluation/plot/roc.py [hunk 1]
```diff
@@ -39,7 +39,14 @@ def roc(y_true, y_score, ax=None):
...
 
     """
-    r = ROC(y_true, y_score, ax=ax)
+    if y_true is not None and y_score is not None:
+        warn(
+            "ROC will change its signature in version 0.10"
+            ", please use ROC.from_raw_data",
+            FutureWarning,
+            stacklevel=2,
+        )
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/roc.py:48]: we no longer need this warning. We'll still support the functional API so there's no need for a warning and the `ROC.__init__` with pass `y_true`, `y_score` has only been out for two days so I doubt people used it.

File: src/sklearn_evaluation/plot/roc.py [hunk 2]
```diff
@@ -193,16 +193,13 @@ class ROC(Plot):
         positive rate of predictions with score >= `thresholds[i]`. If
         None, it will be calculated based on y_true and y_score.
 
+    roc_rates_n_classes : list of dictionaries with 'fpr' and 'tpr' keys
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/roc.py:196]: I'm a bit confused as to why we need this extra parameter. In the [old implementation](https://github.com/ploomber/sklearn-evaluation/blob/6b5f41128fa563c21e5ca7ba0201ab2a0bcea6bb/src/sklearn_evaluation/plot/roc.py#L10), the user only needed to provide y_true and y_score, then we'd figure out the rest (for binary and multi-class). Why do we need this extra parameter?

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by idomic** [comment]
@yafimvo please address

**Comment by yafimvo** [comment]
@idomic let's move [this conversation ](https://github.com/ploomber/sklearn-evaluation/issues/121) here.

I don't think we need to combine issue #98 with this since this one has a lot of changes.

**Review by idomic — APPROVED** [review]
looks good to me

**Comment by idomic** [comment]
Is there documentation? That's the only thing I was missing.
Good test cases!

@edublancas if there's any comments now's the time.

**Comment by idomic** [comment]
@yafimvo I also don't see a changelog entry.

File: src/sklearn_evaluation/plot/roc.py [hunk 3]
```diff
@@ -4,11 +4,10 @@
 from sklearn.preprocessing import label_binarize
 from ..telemetry import SKLearnEvaluationLogger
 from ..util import is_column_vector, is_row_vector
-from ..plot.plot import Plot
 from sklearn_evaluation import __version__
 import json
 from pathlib import Path
-from warnings import warn  # noqa
+from ..plot.plot import AbstractPlot, AbstractComposedPlot
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/roc.py:10]: we're no longer using relative imports, please change

File: src/sklearn_evaluation/plot/roc.py [hunk 4]
```diff
@@ -154,57 +126,51 @@ class ROCAdd(Plot):
...
-    Notes
-    -----
-    It is assumed that the y_score parameter columns are in order.
-    For example, if ``y_true = [2, 2, 1, 0, 0, 1, 2]``, then the
-    first column in y_score must contain the scores for class 0,
-    second column for class 1 and so on.
-
     Examples
     --------
     .. plot:: ../examples/roc_new_api.py
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/roc.py:176]: let's update this example so it shows how the new API works (using from_raw_data).

- roc for binary classification
- roc for multi-class
- comparing two binary classifiers (plot1 + plot2)

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sklearn_evaluation/plot/roc.py (hunk 3 above)
Comment (yafimvo) [reply src/sklearn_evaluation/plot/roc.py]: fixed

File: src/sklearn_evaluation/plot/roc.py (hunk 4 above)
Comment (yafimvo) [reply src/sklearn_evaluation/plot/roc.py]: We already had these examples (except the multi-class).
I changed the names and added an example for multi-class.

please check if these examples are ok.

@idomic Does this multi-class example address #98 ?

Comment (idomic) [reply src/sklearn_evaluation/plot/roc.py]: Not really, the issue there is the error message.

**Pull request merged**